src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/net/NetworkRequestQueue.java

```diff
@@ -0,0 +1,51 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.net;
+
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * Class that represents a custom bounded and concurrent queue.
+ */
+public class NetworkRequestQueue<T> {
+
+  private static final int DEFAULT_MAX_Q_SIZE = 200;
```

Comment:
Just carried over the same number that was being used previously. On a 200 node cluster, it can buffer one flow unit from all nodes. I have made it configurable now though.